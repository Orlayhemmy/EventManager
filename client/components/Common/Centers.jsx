import React from 'react';
import { connect } from 'react-redux';
import { Link, Redirect, browserHistory } from 'react-router-dom';
import swal from 'sweetalert2';
import PropTypes from 'prop-types';
import isEmpty from 'lodash/isEmpty';
import DeleteModal from '../Modal/Container/deleteModal';
import { getAdminActivity } from '../../actions/adminActivityActions';
import { getCenters } from '../../actions/centerActions';

/**
 * @description DisplayCenters form component
 */
export class DisplayCenters extends React.Component {
  state = {
    centerId: '',
    centerName: ''
  };

  onSelect = e => {
    this.setState({
      centerId: e.target.id,
      centerName: e.target.parentNode.id
    });
  };
  /**
   * @memberof DisplayCenters
   * @method componentWillMount
   * @description it gets the necessary object before component is mounted
   */
  componentWillMount() {
    const { searchState } = this.props;
    this.props.getCenters(searchState, searchState.counter);
    this.props.getAdminActivity();
  }

  /**
   * @memberof DisplayCenters
   * @method componentDidUpdate
   * @description it gets the necessary object before component is mounted
   */
  componentDidUpdate() {
    if (this.props.eventCenter.status === 200) {
      swal('Center deleted successfully');
      $(document).ready(function() {
        $('#deleteModal').modal('hide');
      });
    }
  }

  /**
   * @memberof DisplayCenters
   * @method render
   * @description it renders the component
   * @returns the HTML of displaycenters component
   */

  render() {
    const { path } = this.props;
    const { centers } = this.props.eventCenter;
    const { activities } = this.props.adminActivity;
    let adminCenter;
    const recentActivity = activities.map((activity, index) => {
      const creationDate = activity.createdAt
        .replace(/-/g, '/')
        .replace('Z', '')
        .replace('T', ' ')
        .slice(0, 16);
      return (
        <div key={index} className="main-color-bg p-2 mb-1">
          <Link to="/view-center-event">
            <p
              className="activity-font mb-0 p-1 fw"
              onClick={this.showCenter}
              id={activity.centerId}
            >
              {activity.description}
              <br />
              {creationDate}
            </p>
          </Link>
        </div>
      );
    });
    if (isEmpty(this.props.eventCenter.centers)) {
      adminCenter = (
        <div className="emptyCenter img-fluid text-center ml-2 mt-2">
          <span>
            <p className="display-3">No Center Found</p>
          </span>
        </div>
      );
    } else {
      adminCenter = centers.map((center, index) => {
        return (
          <div className="row card reverse p-4" key={index}>
            <div className="col-lg-4 col-md-12 col-sm-12 text-center">
              <img className="img-fluid" src={center.imageUrl} />
            </div>
            <div className="col-8 col-md-8 col-sm-12 pl-4">
              <h2 className="media-heading text-center">
                <Link to="/view-center-event">
                  <span onClick={this.props.showCenter} id={center.id}>
                    {center.centerName}
                  </span>
                </Link>
              </h2>

              <h3>
                <span>Location: </span> {center.location}
              </h3>

              <h3>
                <span>capacity: </span> {center.capacity}
              </h3>

              <h3>
                <span>facilities: </span> {center.facilities}
              </h3>

              <h3>
                <span>cost: </span> {center.cost}
              </h3>

              <h3>
                <span>description: </span> {center.description}
              </h3>
            </div>
            <span
              className="trash p-2"
              data-toggle="modal"
              data-target="#deleteModal"
              id={center.centerName}
              onClick={this.onSelect}
            >
              <i id={center.id} className="fa fa-trash trash" />
            </span>
          </div>
        );
      });
    }

    const adminCenterPage = (
      <div className="section">
        <div className="row wc">
          <div className="col-lg-8 mr-3">
            {this.props.counter > 0 ? (
              <div className="pagination-icon">
                <i
                  className="fa fa-chevron-up icon"
                  id="previous"
                  onClick={this.props.nextCenters}
                />
                <p class="tooltiptext">Previous Centers</p>
              </div>
            ) : (
              ''
            )}
            {this.props.eventCenter.isNext ? (
              <div className="pagination-icon mt-y1-5">
                <i
                  className="fa fa-chevron-down icon"
                  id="next"
                  onClick={this.props.nextCenters}
                />
                <p class="tooltiptext">Next Centers</p>
              </div>
            ) : (
              ''
            )}
            {adminCenter}
          </div>
          <div className="col-lg-3 card p-2 ho">
            <h2>Notifications</h2>
            {recentActivity}
          </div>
        </div>
      </div>
    );
    const guestCenters = centers.map((center, index) => {
      return (
        <div className="row card reverse p-4" id={center.id} key={index}>
          <div className="col-lg-4 col-md-12 col-sm-12 text-center">
            <img className="img" src={center.imageUrl} />
          </div>
          <div className="col-8 col-md-8 pl-4">
            <h2 className="media-heading text-center">{center.centerName}</h2>

            <h3>
              <span>Location: </span> {center.location}
            </h3>

            <h3>
              <span>capacity: </span> {center.capacity}
            </h3>

            <h3>
              <span>facilities: </span> {center.facilities}
            </h3>

            <h3>
              <span>Cost: </span> {center.cost}
            </h3>

            <h3>
              <span>description: </span> {center.description}
            </h3>
          </div>
        </div>
      );
    });

    const guestCenterPage = (
      <div>
        {this.props.counter > 0 ? (
          <div className="pagination-icon">
            <i
              className="fa fa-chevron-up icon"
              id="previous"
              onClick={this.props.nextCenters}
              data-toggle="tooltip"
              data-placement="right"
              title="Previous Centers"
            />
          </div>
        ) : (
          ''
        )}
        {centers.length > 4 ? (
          <div className="pagination-icon mt-y1-5">
            <i
              className="fa fa-chevron-down icon"
              id="next"
              onClick={this.props.nextCenters}
            />
            <p class="tooltiptext">Filter Centers</p>
          </div>
        ) : (
          ''
        )}
        {guestCenters}
      </div>
    );
    return (
      <div>
        <div className="search-icon" onClick={this.props.searchNav}>
          <i className="fa fa-search" />
          <p class="tooltiptext">Filter Centers</p>
        </div>
        {this.props.auth.user.isAdmin ? adminCenterPage : guestCenterPage}
        <DeleteModal path={path} centerProps={this.state} />
      </div>
    );
  }
}
const propTypes = {
  auth: PropTypes.object.isRequired,
  eventCenter: PropTypes.object.isRequired,
  adminActivity: PropTypes.object.isRequired,
  getAdminActivity: PropTypes.func.isRequired,
  getCenters: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  eventCenter: state.center,
  auth: state.auth,
  adminActivity: state.adminActivity
});

DisplayCenters.propTypes = propTypes;

export default connect(
  mapStateToProps,
  {
    getAdminActivity,
    getCenters
  }
)(DisplayCenters);
