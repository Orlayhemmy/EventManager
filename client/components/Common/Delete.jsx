import React from 'react';

/**
 * @description Delete component
 */
export default class Delete extends React.Component {
  render() {
    const { title, onDelete, onCancel } = this.props;
    return (
      <div id="delete-content">
        <p>Are you sure you want to delete {title}?</p>
        <i id="yes" className="fa fa-trash red" onClick={onDelete}/>
        <i id="no" className="fa fa-save green" onClick={onCancel}/>
        <br/>
        <span>
          <br />Yes
        </span>
        <span>
          <br />No
        </span>
      </div>
    );
  }
}
