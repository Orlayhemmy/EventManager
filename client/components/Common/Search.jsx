import React from 'react';

/**
 * @description SearchForm  component
 */
export default class SearchForm extends React.Component {
  /**
   * @memberof SearchForm
   * @method render
   * @description it renders the component
   * @returns the HTML of SearchForm
   */
  render() {
    const greater = ' >';
    const lesser = ' <';
    const equal = ' =';
    const between = '<>';
    const { location, facilities, capacity, capacityType, btwValue, errors } = this.props.criteria;
    return (
      <div className="row bw p-3" id="center-search">
        <p className="subtitle">
          <i className="fa fa-filter green" /> filter centers by
        </p>

        <div className="input-group">
          <div class="input-group-prepend">
            <span class="input-group-text">
              <i className="fa fa-map-marker" />
            </span>
          </div>
          <input
            id="location"
            value={location}
            placeholder="location"
            type="text"
            onChange={this.props.onChange}
            className="form-control"
          />
          <div className="help-block">{errors.location}</div>

          <div class="input-group-prepend">
            <span class="input-group-text">
              <i className="fa fa-cog" />
            </span>
          </div>
          <input
            id="facilities"
            value={facilities}
            placeholder="facilities"
            type="text"
            onChange={this.props.onChange}
            className="form-control"
          />
          <div className="help-block">{errors.facilities}</div>
            <div class="input-group-prepend">
            <span class="input-group-text">
              <select onChange={this.props.onChange} id="capacityType">
                <option value="greater">{greater}</option>
                <option value="lesser">{lesser}</option>
                <option value="equal">{equal}</option>
                <option value="between">{between}</option>
              </select>
            </span>
          </div>
          <input
            id="capacity"
            value={capacity}
            placeholder="capacity"
            type="number"
            onChange={this.props.onChange}
            className="form-control"
          />
          <input
            id="btwValue"
            value={btwValue}
            placeholder="capacity"
            type="number"
            onChange={this.props.onChange}
            className="form-control"
            hidden
          />
          <button className="btn btn-success" onClick={this.props.search}>
            <i className="fa fa-search">
              {' '}
              Search
            </i>
          </button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  center: state.center
});
