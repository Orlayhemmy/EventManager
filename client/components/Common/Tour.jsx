import React from 'react'; // eslint-disable-line

const tour = props => (
  <div id="tour" className="tour pagination-icon w-auto" onClick={props.tour}>
    <i className="fa fa-bus-alt" />
    <p class="tooltiptext">Take a tour</p>
  </div>
);
export default tour;
