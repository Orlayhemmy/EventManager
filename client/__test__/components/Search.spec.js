import React from 'react';
import { SearchForm } from '../../components/Common/Search';

describe('Search Form Component', () => {
  const getCenters = jest.fn();
  const criteria = {
    location: '',
    facilities: '',
    capacity: '',
    btwValue: '',
    errors: '',
    capacityType: ''
  };
  const criteria2 = {
    location: '',
    facilities: '',
    capacity: '',
    btwValue: '',
    errors: '',
    capacityType: 'between'
  };
  const search = jest.fn();
  const onChange = jest.fn();

  it('should render component without error', () => {
    const wrapper = renderer
      .create(<SearchForm criteria={criteria} search={search} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component without error', () => {
    const wrapper = shallow(<SearchForm criteria={criteria2} search={search} onChange={onChange} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should reset state ', () => {
    const wrapper = shallow(<SearchForm criteria={criteria2} search={search} onChange={onChange} getCenters={getCenters}/>);
    const button = (wrapper).find('#reset');
    button.simulate('click', {
      preventDefault: () => {}
    });
  });
});
