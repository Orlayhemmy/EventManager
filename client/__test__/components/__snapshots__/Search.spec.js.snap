// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should render component without error 1`] = `
<div
  className="search-nav"
  id="search-nav"
>
  <div
    className="search-field"
  >
    <p
      className="subtitle"
    >
      <i
        className="fa fa-filter green"
      />
       filter centers by
    </p>
    <div
      className="input-group pb-2 pt-3"
    >
      <div
        className="help-block"
      />
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          <i
            className="fa fa-map-marker"
          />
        </span>
      </div>
      <input
        className="form-control"
        id="location"
        onChange={undefined}
        placeholder="location"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group pb-2"
    >
      <div
        className="help-block"
      />
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          <i
            className="fa fa-cog"
          />
        </span>
      </div>
      <input
        className="form-control"
        id="facilities"
        onChange={undefined}
        placeholder="facilities"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group pb-2"
    >
      <select
        className="form-control"
        id="capacityType"
        onChange={undefined}
      >
        <option>
          capacity conditions
        </option>
        <option
          value="greater"
        >
          greater &gt;
        </option>
        <option
          value="lesser"
        >
          lesser &lt;
        </option>
        <option
          value="equal"
        >
          equal =
        </option>
        <option
          value="between"
        >
          between &lt;&gt;
        </option>
      </select>
    </div>
    <div
      className="input-group pb-2"
      id="otherConditions"
    >
      <div
        className="help-block"
      />
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          <i
            className="fa fa-users"
          />
        </span>
      </div>
      <input
        className="form-control"
        id="capacity"
        onChange={undefined}
        placeholder="capacity"
        type="number"
        value=""
      />
    </div>
    <div
      className="pb-2"
      hidden={true}
      id="btwValue"
    >
      <div
        className="input-group"
      >
        <div
          className="help-block"
        />
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
          >
            <i
              className="fa fa-users"
            />
          </span>
        </div>
        <input
          className="form-control"
          id="btwValue"
          onChange={undefined}
          placeholder="Capacity"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="input-group pt-2"
      id="submit"
    >
      <button
        className="btn btn-success"
        id="submit"
        onClick={[MockFunction]}
        type="button"
      >
        <i
          className="fa fa-search"
        >
           Search
        </i>
      </button>
      <button
        className="btn btn-danger fr"
        id="reset"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-sync-alt"
        >
           Reset
        </i>
      </button>
    </div>
  </div>
</div>
`;

exports[`Search Component should render component without error 2`] = `
<div
  className="search-nav"
  id="search-nav"
>
  <div
    className="search-field"
  >
    <p
      className="subtitle"
    >
      <i
        className="fa fa-filter green"
      />
       filter centers by
    </p>
    <div
      className="input-group pb-2 pt-3"
    >
      <div
        className="help-block"
      />
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          <i
            className="fa fa-map-marker"
          />
        </span>
      </div>
      <input
        className="form-control"
        id="location"
        onChange={undefined}
        placeholder="location"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group pb-2"
    >
      <div
        className="help-block"
      />
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          <i
            className="fa fa-cog"
          />
        </span>
      </div>
      <input
        className="form-control"
        id="facilities"
        onChange={undefined}
        placeholder="facilities"
        type="text"
        value=""
      />
    </div>
    <div
      className="input-group pb-2"
    >
      <select
        className="form-control"
        id="capacityType"
        onChange={undefined}
      >
        <option>
          capacity conditions
        </option>
        <option
          value="greater"
        >
          greater &gt;
        </option>
        <option
          value="lesser"
        >
          lesser &lt;
        </option>
        <option
          value="equal"
        >
          equal =
        </option>
        <option
          value="between"
        >
          between &lt;&gt;
        </option>
      </select>
    </div>
    <div
      className="input-group pb-2"
      id="otherConditions"
    >
      <div
        className="help-block"
      />
      <div
        className="input-group-prepend"
      >
        <span
          className="input-group-text"
        >
          <i
            className="fa fa-users"
          />
        </span>
      </div>
      <input
        className="form-control"
        id="capacity"
        onChange={undefined}
        placeholder="capacity"
        type="number"
        value=""
      />
    </div>
    <div
      className="pb-2"
      hidden={true}
      id="btwValue"
    >
      <div
        className="input-group"
      >
        <div
          className="help-block"
        />
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
          >
            <i
              className="fa fa-users"
            />
          </span>
        </div>
        <input
          className="form-control"
          id="btwValue"
          onChange={undefined}
          placeholder="Capacity"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      className="input-group pt-2"
      id="submit"
    >
      <button
        className="btn btn-success"
        id="submit"
        onClick={[MockFunction]}
        type="button"
      >
        <i
          className="fa fa-search"
        >
           Search
        </i>
      </button>
      <button
        className="btn btn-danger fr"
        id="reset"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-sync-alt"
        >
           Reset
        </i>
      </button>
    </div>
  </div>
</div>
`;
